File: C++/Builder/LaptopManufacturer/laptop.cc
#include <iostream>
#include "laptop.h"

void Laptop::set_name(const std::string &name) {
    _name = name;
}

void Laptop::set_cpu(const std::string &cpu) {
    _cpu = cpu;
}

void Laptop::set_memory(const std::string &memory) {
    _memory = memory;
}

void Laptop::set_os(const std::string &os) {
    _os = os;
}

void Laptop::show_specs() const {
    std::cout << std::string(50, '=') << "\n"
              << "This is a " << _name << ",\n"
              << "with a " << _cpu << " cpu,\n"
              << "with " << _memory << " of memory,\n"
              << "running a " << _os << " operating system.\n\n";


}

File: Java/Builder/LaptopManufacturer/src/com/ericgtkb/Laptop.java
package com.ericgtkb;

public class Laptop {
    private String name;
    private String cpu;
    private String memory;
    private String os;

    public void setName(String name) {
        this.name = name;
    }

    public void setCpu(String cpu) {
        this.cpu = cpu;
    }

    public void setMemory(String memory) {
        this.memory = memory;
    }

    public void setOs(String os) {
        this.os = os;
    }

    public void showSpecs() {
        System.out.println("================================================");
        System.out.println("This is a " + name + ",");
        System.out.println("with a " + cpu + " cpu,");
        System.out.println("with " + memory + " of memory,");
        System.out.println("running a " + os + " operating system.");
        System.out.println();
    }
}
